botx/models/attachments.py

```diff
@@ -0,0 +1,262 @@
+"""Module with attachments for botx."""
+
+from typing import List, Optional, Union
+
+from pydantic import BaseModel
+
+from botx.exceptions import MissingAttributeError
+from botx.models.enums import AttachmentsTypes
+
+MAIL_PROTO_STR = "mailto:"
+TELEPHONE_PROTO_STR = "tel://"
+
+
+class FileData(BaseModel):
+    """Class that represents file in RFC 2397 format."""
+
+    #: name of file.
+    file_name: Optional[str]
+
+    #: file content in RFC 2397 format.
+    content: str  # noqa: WPS110
+
+
+class Image(FileData):
+    """Image model from botx."""
+
+    file_name: str = "image.jpg"
+
+
+class Video(FileData):
+    """Image model from botx."""
+
+    file_name: str = "video.mp4"
+    #: video duration
+    duration: int
+
+
+class Document(FileData):
+    """Document model from botx."""
+
+    file_name: str = "document.docx"
+
+
+class Voice(BaseModel):
+    """Voice model from botx."""
+
+    #: file content in RFC 2397 format.
+    content: str  # noqa: WPS110
+
+    #: voice duration
+    duration: int
+
+
+class Location(BaseModel):
+    """Location model from botx."""
+
+    #: name of location
+    location_name: str
+
+    #: address of location
+    location_address: str
+
+    #: latitude of location
+    location_lat: float
+
+    #: longitude of location
+    location_lng: float
+
+
+class Contact(BaseModel):
+    """Contact model from botx."""
+
+    #: name of contact
+    contact_name: str
+
+
+class Link(BaseModel):
+    """Class that marked as Link from botx."""
+
+    #: url of link
+    url: str
+
+    #: title of url
+    url_title: Optional[str] = None
+
+    #: link on preview of this link
+    url_preview: Optional[str] = None
+
+    #: text on preview
+    url_text: Optional[str] = None
+
+    def is_mail(self) -> bool:
+        """Confirm that is email link."""
+        return self.url.startswith(MAIL_PROTO_STR)
+
+    def is_telephone(self) -> bool:
+        """Confirm that is telephone link."""
+        return self.url.startswith(TELEPHONE_PROTO_STR)
+
+    def is_link(self) -> bool:
+        """Confirm that is link on resource."""
+        return not (self.is_mail() or self.is_telephone())
+
+    @property
+    def mailto(self) -> str:
+        """Property that retuning email address without protocol."""
+        if not self.is_mail():
+            raise MissingAttributeError(attribute_name="mailto")
+        # fmt: off
+        return self.url[len(MAIL_PROTO_STR):]
+        # fmt: on
```

Comment:
ignore error for space after colon either on this line only or for entire project. it's not so important when code formatting is controlled by black